### PR TITLE
Reduce runtime of OptimizationExample

### PR DESCRIPTION
**Pull request by tkuchida**
Addresses #394

### CODE REVIEW

File: OpenSim/Examples/OptimizationExample_Arm26/OptimizationExample.cpp [hunk 1]
```diff
@@ -131,10 +135,11 @@ int main()
         const CoordinateSet& coords = osimModel.getCoordinateSet();
         coords.get("r_shoulder_elev").setValue(si, -1.57079633);
 
-        // Set the initial muscle activations 
+        // Set the initial muscle activations and make all tendons rigid.
         const Set<Muscle> &muscleSet = osimModel.getMuscles();
-        for(int i=0; i< muscleSet.getSize(); i++ ){
+        for(int i=0; i<muscleSet.getSize(); ++i) {
```
Comment (chrisdembia) [inline OpenSim/Examples/OptimizationExample_Arm26/OptimizationExample.cpp]: ++1

**Comment by chrisdembia** [comment]
testOptimizationExample fails when comparing the results of the optimization to those stored in [this file](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Tests/OptimizationExample_Arm26/std_Arm26_noActivation_states.sto). Perhaps the file should just be updated, if the new results look reasonable?

**Comment by aseth1** [comment]
>testOptimizationExample fails when comparing the results of the optimization to those stored in this file. Perhaps this file should just be updated, if the new results look reasonable?

@tkuchida did you update the standard on your machine for the test to pass?

**Comment by tkuchida** [comment]
>Perhaps this file should just be updated, if the new results look reasonable?

In the old standard motion, the elbow hyperextends to -28° at t=0.25s. With rigid tendons, the elbow remains relatively straight, hyperextending to only -4.6° at t=0.25s.

**Comment by tkuchida** [comment]
Runtime of testOptimizationExampleRuns:

compiler | master | this branch
:---: | :---: | :---:
MSVC | 411.87s | 123.47s
clang | 539.17s | 132.28s
gcc | 788.05s | 227.29s

**Comment by sherm1** [comment]
:+1:

**Comment by chrisdembia** [comment]
This is somewhat not related to this PR, but is the pennation angle functionality that was previously tested through this test now being tested elsewhere?

Also, for this maximal task, I would have thought the control inputs would all either be 0.01 or 0.99-ish. However, two of the control inputs are somewhere in the middle.

**Comment by chrisdembia** [comment]
Would it make sense to use an initial guess that is closer to the solution?

**Comment by tkuchida** [comment]
>is the pennation angle functionality that was previously tested through this test now being tested elsewhere?

There's a test ensuring that the pennation model is operating correctly in isolation (testMuscleFixedWidthPennationModel), but there's no test ensuring that the pennation model is operating correctly once hooked up to a Millard2012EquilibriumMuscle. I've added this to the list of tasks in #366.

**Comment by aseth1** [comment]
The test that optimal solution is within 20% of the reference controls remains commented out. We should uncomment to make sure we are getting similar solutions on all platforms. 
@chrisdembia that is a great and simple solution. It may also results in more uniform solutions on the different OS' and not to mention whittle down unnecessary computation time for the test case.

**Comment by tkuchida** [comment]
>that is a great and simple solution.

@aseth1 Are you referring to the suggestion to use an initial guess that is closer to the solution?

**Comment by tkuchida** [comment]
Runtime of testOptimizationExampleRuns:

compiler | master | previous push | now
:---: | :---: | :---: | :---:
MSVC | 411.87s | 123.47s | 76.57s
clang | 539.17s | 132.28s | 105.02s
gcc | 788.05s | 227.29s | 103.20s

**Comment by sherm1** [comment]
> Runtime of testOptimizationExampleRuns

Can you post that again? I blinked and missed it.

**Comment by tkuchida** [comment]
New runtime of testOptimizationExampleRuns:

compiler | master | this branch
:---: | :---: | :---:
MSVC | 411.87s | 63.24s
clang | 539.17s | 99.59s
gcc | 788.05s | 111.42s

The test now (1) runs faster, and (2) checks the results of a simulation using the _optimal_ controls rather than the _initial_ controls.

**Comment by aseth1** [comment]
Looks good to me including visualizing the results in OpenSim. Thanks @tkuchida .

**Pull request merged**